src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java

```diff
@@ -0,0 +1,102 @@
+package cz.muni.pa165.teamwhite.formula1;
+
+import cz.muni.pa165.teamwhite.formula1.dao.UserDao;
+import cz.muni.pa165.teamwhite.formula1.entity.User;
+import cz.muni.pa165.teamwhite.formula1.enums.Role;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.persistence.*;
+import javax.transaction.Transactional;
+import javax.validation.ConstraintViolationException;
+import java.util.List;
+
+/**
+ * @author Jiří Andrlík
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class UserTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceUnit
+    EntityManagerFactory emf;
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Autowired
+    private UserDao userDao;
+
+
+    @Test
+    public void testCreate(){
+        User kaja = new User("kaja", "CoTeToZajima", Role.MANAGER);
+        userDao.create(kaja);
+        Assert.assertTrue(true);
+    }
+
+    @Test
+    public void testFetchById(){
+        User jiri = new User("andrlos", "1234", Role.MANAGER);
+        userDao.create(jiri);
+        User u = userDao.findById(jiri.getId());
+        Assert.assertEquals(u.getLogin(), jiri.getLogin());
+        Assert.assertEquals(u.getPassword(), jiri.getPassword());
+        Assert.assertEquals(u.getRole(), jiri.getRole());
+    }
+
+    @Test
+    @Rollback
+    public void testFetchByNameAndRemove(){
+        User tom = new User("lime", "4321", Role.ENGINEER);
+        userDao.create(tom);
+        tom = userDao.findByLogin("lime");
+        userDao.remove(tom);
+        Assert.assertThrows(NoResultException.class, () -> userDao.findByLogin("lime"));
```

Comment:
expectedExceptions?

Reply:
I like this maybe even more than the annotation :thinking:... you are explicitly stating where you are expecting the exception...even if it is not as elegant as the annotation

Reply:
ok, but at least we could use same "style" in one file

Reply:
Definitelly...

Reply:
solved :D

Reply:
:smile: did not expect that... but, why not :smile: Someday in the feature, you can rewrite all other tests to look this way :trollface: